tests/Operation/WatchFunctionalTest.php

```diff
@@ -1010,18 +1026,22 @@ public function testStartAfterOption()
         $this->insertDocument(['_id' => 1, 'x' => 'foo']);
         $this->insertDocument(['_id' => 2, 'x' => 'bar']);
 
-        $changeStream->next();
-        $this->assertTrue($changeStream->valid());
+        $this->advanceCursorUntilValid($changeStream);
 
         $resumeToken = $changeStream->current()->_id;
 
         $options = $this->defaultOptions + ['startAfter' => $resumeToken];
         $operation = new Watch($this->manager, $this->getDatabaseName(), $this->getCollectionName(), [], $options);
         $changeStream = $operation->execute($this->getPrimaryServer());
-        $this->assertSame($resumeToken, $changeStream->getResumeToken());
+        $this->assertEquals($resumeToken, $changeStream->getResumeToken());
```

Comment:
Was this change actually necessary to resolve a failure or is this just a preemptive change?

I believe `assertSame()` would have worked here, since the spec calls for initializing the resume token from the corresponding option and [`Watch::getInitialResumeToken()`](https://github.com/mongodb/mongo-php-library/blob/e0a2b89974139834e1737f658f58e7f77c6314ae/src/Operation/Watch.php#L339) returns the option directly. The only way I imagine `assertEquals()` to be necessary here is if the `startAfter` option was also returned via the `postBatchResumeToken` of the initial, empty response from `aggregate`. But even if a sharded cluster returned an empty `aggregate` response for some reason (e.g. waiting for periodic noop), I don't see how it'd return our resume token without also include the very same document from which we pulled it.

On a separate note: This test closely mirrors `testResumeAfterOption()`, which doesn't assert that `$changeStream->getResumeToken()` matches the `resumeAfter` option provided to `watch()`. It only asserts that the first document returned is what we expect. I don't object to asserting that the initial resume token is also what we expect, but I'd propose you also make the same change to `testResumeAfterOption()` for consistency. Lastly, if `assertSame()` really doesn't work here I'd suggest using `assertSameDocument()` over `assertEquals()`, since we use that in various other places throughout this test file.

Reply:
This change was necessary, as the initial aggregate command returns a `postBatchResumeToken`, which always takes precedence over `startAfter` or `resumeAfter` options. While this resume token has the same value as the one in the `startAfter` option, it is not the same object as it was returned by the database. Hence the change to `assertEquals` instead of `assertSame`.